src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordScriptureFragment.kt

```diff
@@ -1,86 +1,33 @@
 package org.wycliffeassociates.otter.jvm.app.ui.takemanagement.view
 
 import com.github.thomasnield.rxkotlinfx.toObservable
-import com.jfoenix.controls.JFXSnackbar
 import de.jensd.fx.glyphs.materialicons.MaterialIcon
 import de.jensd.fx.glyphs.materialicons.MaterialIconView
-import javafx.application.Platform
-import javafx.beans.property.SimpleObjectProperty
-import javafx.event.EventHandler
 import javafx.geometry.Pos
-import javafx.scene.Node
-import javafx.scene.control.ButtonType
 import javafx.scene.control.ContentDisplay
-import javafx.scene.input.MouseEvent
-import javafx.scene.layout.FlowPane
 import javafx.scene.layout.Priority
-import javafx.scene.layout.StackPane
-import javafx.scene.layout.VBox
 import org.wycliffeassociates.otter.common.data.workbook.Take
 import org.wycliffeassociates.otter.jvm.app.theme.AppStyles
-import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.TakeContext
 import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel.RecordScriptureViewModel
-import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel.AudioPluginViewModel
-import org.wycliffeassociates.otter.jvm.app.widgets.progressdialog.progressdialog
-import org.wycliffeassociates.otter.jvm.app.widgets.takecard.OldTakeCard
-import org.wycliffeassociates.otter.jvm.app.widgets.takecard.oldtakecard
+import org.wycliffeassociates.otter.jvm.app.widgets.takecard.*
 import tornadofx.*
 
-class RecordScriptureFragment : Fragment() {
-    private val audioPluginViewModel: AudioPluginViewModel by inject()
+private class RecordableViewModelProvider: Component() {
     private val recordScriptureViewModel: RecordScriptureViewModel by inject()
-    private val recordableViewModel = recordScriptureViewModel.recordableViewModel
+    fun get() = recordScriptureViewModel.recordableViewModel
+}
 
-    // The currently selected take
-    private var selectedTakeProperty = SimpleObjectProperty<OldTakeCard>()
-    // Take at the top to compare to an existing selected take
-    private var draggingTakeProperty = SimpleObjectProperty<OldTakeCard>()
+class RecordScriptureFragment
+    : RecordableFragment(RecordableViewModelProvider().get()) {
```

Comment:
could you use di scopes here instead of the get?

Reply:
I looked into this, but I haven't figured out how to make it work for this situation